non-fungible-token/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+		/// Token existence check by owner and class ID.
+		pub TokensByOwner get(fn tokens_by_owner): double_map hasher(twox_64_concat) T::AccountId, hasher(twox_64_concat) (T::ClassId, T::TokenId) => Option<()>;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+	}
+}
+
+impl<T: Trait> Module<T> {
+	/// Create NFT(non fungible token) class
+	pub fn create_class(owner: &T::AccountId, metadata: CID, data: T::ClassData) -> Result<T::ClassId, DispatchError> {
+		let class_id = Self::next_class_id();
+		ensure!(class_id != T::ClassId::max_value(), Error::<T>::NoAvailableClassId);
+
+		let info = ClassInfo {
+			metadata,
+			total_issuance: Default::default(),
+			owner: owner.clone(),
+			data,
+		};
+		Classes::<T>::insert(class_id, info);
+		NextClassId::<T>::mutate(|id| *id += 1.into());
+
+		Ok(class_id)
+	}
+
+	/// Transfer NFT(non fungible token) from `from` account to `to` account
+	pub fn transfer(from: &T::AccountId, to: &T::AccountId, token: (T::ClassId, T::TokenId)) -> DispatchResult {
+		ensure!(Tokens::<T>::contains_key(token.0, token.1), Error::<T>::TokenNotFound);
+		ensure!(
+			TokensByOwner::<T>::contains_key(from.clone(), token),
+			Error::<T>::NoPermission
+		);
+
+		if from == to {
+			return Ok(());
+		}
+
+		Tokens::<T>::mutate(token.0, token.1, |token_info| {
+			if let Some(info) = token_info {
+				info.owner = to.clone();
+			}
+		});
+		TokensByOwner::<T>::remove(from.clone(), token);
+		TokensByOwner::<T>::insert(to.clone(), token, ());
+
+		Ok(())
+	}
+
+	/// Mint NFT(non fungible token) to `owner`
+	pub fn mint(
+		owner: &T::AccountId,
+		class_id: T::ClassId,
+		metadata: CID,
+		data: T::TokenData,
+	) -> Result<T::TokenId, DispatchError> {
+		let token_id = Self::next_token_id();
+		ensure!(token_id != T::TokenId::max_value(), Error::<T>::NoAvailableTokenId);
+		let token_info = TokenInfo {
+			metadata,
+			owner: owner.clone(),
+			data,
+		};
+		Classes::<T>::try_mutate(class_id, |class_info| -> DispatchResult {
+			if let Some(info) = class_info {
+				info.total_issuance = info
+					.total_issuance
+					.checked_add(&1.into())
+					.ok_or(Error::<T>::NumOverflow)?;
+			}
+			Ok(())
+		})?;
+		Tokens::<T>::insert(class_id, token_id, token_info);
+		TokensByOwner::<T>::insert(owner.clone(), (class_id, token_id), ());
+
+		Ok(token_id)
+	}
+
+	/// Burn NFT(non fungible token) from `owner`
+	pub fn burn(owner: &T::AccountId, token: (T::ClassId, T::TokenId)) -> DispatchResult {
+		ensure!(Tokens::<T>::contains_key(token.0, token.1), Error::<T>::TokenNotFound);
+		ensure!(
+			TokensByOwner::<T>::contains_key(owner.clone(), token),
+			Error::<T>::NoPermission
+		);
+
+		Classes::<T>::try_mutate(token.0, |class_info| -> DispatchResult {
+			if let Some(info) = class_info {
+				info.total_issuance = info
+					.total_issuance
+					.checked_sub(&1.into())
+					.ok_or(Error::<T>::NumOverflow)?;
+			}
+			Ok(())
+		})?;
+		Tokens::<T>::remove(token.0, token.1);
```

Comment:
Should be able to use `take().ok_or(Error::<T>::TokenNotFound)` to combine remove and check.